ouroboros-consensus/ouroboros-consensus.cabal

```diff
@@ -331,9 +319,9 @@ library
 
                      , io-classes       ^>=0.3
                      , typed-protocols
-                     , network-mux
-                     , ouroboros-network-framework ^>=0.2
-                     , ouroboros-network ^>=0.2
```

Comment:
Let's remove the upper bounds in this `.cabal` file --- none of our others have them; I think they're appropriate in the `cabal.project` file now, right?

cc @dnadales @amesgen

Reply:
We discussed this --- the plan is the _opposite_. The bounds are only temporarily in `cabal.project`.

My desiderata:

- each upstream package's bound is stated in _one place_ (this makes bumps trivial to do, and clear in diffs)
- there is some logic behind where that is

We had two ideas for this:

- choose one `.cabal` file as the "owner" of each upstream dep (eg `ouroboros-consensus-shelley.cabal` would own the Shelley-related ledger packages)

- reorganize all Consensus packages into a single `.cabal` file and use the multi-library support with reusable stanzas that include the bounded `build-depend`s

We'll address this shortly in a follow-up PR --- it's related to but beyond the scope of this one.

Reply:
See #4185